org.gimp.GIMP.json

```diff
@@ -16,12 +16,34 @@
                     "--talk-name=org.gtk.vfs", "--talk-name=org.gtk.vfs.*",
                     "--talk-name=org.freedesktop.FileManager1"],
     "tags": ["stable"],
-    "cleanup": ["/include", "/lib/pkgconfig", "/lib/cmake/", "/share/pkgconfig",
-                "/share/aclocal", "/man", "/share/man", "/share/gtk-doc", "/share/doc",
-                "/share/vala", "*.la", "*.a", "/bin/wmf*", "/bin/libwmf-*",
-                "/share/pygobject/2.0/codegen",
-                "/bin/pygtk*", "/bin/pygobject*", "/lib/pygtk",
-                "/bin/python*-config", "/bin/pip*", "/bin/smtpd.py" ],
+    "cleanup": [
+        "/lib/cmake/",
+        "/man", "/share/man",
+        "/share/gtk-doc",
+        "/share/doc",
+        "/share/vala",
+        "*.la",
+        "*.a",
+        "/bin/wmf*",
+        "/bin/libwmf-*",
+        "/share/pygobject/2.0/codegen",
+        "/bin/pygtk*",
+        "/bin/pygobject*",
+        "/lib/pygtk",
+        "/bin/python*-config",
+        "/bin/pip*",
+        "/bin/smtpd.py"
+    ],
+    "add-extensions": {
+        "org.gimp.GIMP.Plugin": {
```

Comment:
> For a plug-in manifest to declare itself as a GIMP extension of type "org.gimp.GIMP.Plugin", all it must do is be named "org.gimp.GIMP.Plugin.*"? I am asking since I can't find anywhere a reference to this string in plug-in manifest (except in their own app id prefix). So I want to make sure this is how it works.

This is was the `subdirectories` is for in the extension point definition.
`org.gimp.GIMP.Plugin` is `/app/extensions`
`org.gimp.GIMP.Plugin.GMic` is `/app/extensions/GMic`. And so on.

> 
> Once again if we get a manual extension for instance (second time I give the example, this is because this specific extension point has been asked to us a few times too), I want to make sure a plug-in manifest can easily been distinguished hence end up inside the right extension point.

How do these work I couldn't find the part about "manual" ? I don't mind actually working on this to make sure there is no conflict.

> Also is that what determines what `${FLATPAK_DEST}` ends up to be? I see you use this env variable in plug-in manifests. does it equate to `/app/extensions/` when used inside a GIMP plug-in manifest?

`FLATPAK_DEST` in the exteision is set based on the extension point as defined above. So for GMic it is `/app/extensions/GMic`

Reply:
I forget to mention the required `"build-extension": true` to indicate it is an extesion.

Reply:
FWIW I'm working on a tutorial on how to write extensions for flatpak.

Reply:
> How do these work I couldn't find the part about "manual" ? I don't mind actually working on this to make sure there is no conflict.

Well we don't have a manual extension point yet. Basically GIMP looks in `$PREFIX/share/gimp/2.0/help/`. If it finds a manual which corresponds to currently set locale, it will use it by default (typically when clicking the "Help" button on various dialog or with F1). If it doesn't find the manual for your language, but you have other manuals installed, it will also propose you to select one of the other langs (which can also be done in Preferences).

So yeah basically it would be nice that each manual gets its own extension, so that someone can install just the French manual, or just the Korean one, or whatever (it's quite a lot of data, with images and such).

You can find the manuals at: https://download.gimp.org/pub/gimp/help/

Right now, there is only `gimp-help-2.10.0.tar.bz2` for the 2.10 series. Note that this is the source, not built yet. Since built manual is just static HTML, if you wish to test, you may just install a manual for instance with your distrib package manager (on Fedora, the package for the French manual would be `gimp-help-fr` for instance) and just copy the HTML files over.